README.md
# Duration Picker for flutter

Fork from duration_picker. https://github.com/juliansteenbakker/duration_picker

A little widget for picking durations. Heavily inspired from the Material Design time picker widget.

<img src="https://raw.githubusercontent.com/juliansteenbakker/duration_picker/master/example.gif" height="480px" >

## Example Usage:

```yaml
dependencies:
  duration_picker: "^1.0.0"
```

```dart
import 'package:duration_picker2/duration_picker.dart';
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Duration Picker Demo',
      theme: ThemeData(
        useMaterial3: true,
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Duration Picker Demo'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  Duration _initTime = Duration(seconds: 21);
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Expanded(
                child: DurationPicker(
                  initialTime: _initTime,
                  minTime: const Duration(seconds: 15),
                  maxTime: const Duration(seconds: 30),
                  enableHapticFeedback: false,
                  baseUnit: BaseUnit.second,
                  onChange: (val) {
                    setState(() {
                      _initTime = val;
                    });
                  },
                  snapToMins: 5.0,
                ))
          ],
        ),
      ),
      floatingActionButton: Builder(
          builder: (BuildContext context) => FloatingActionButton(
            onPressed: () async {
              var resultingDuration = await showDurationPicker(
                context: context,
                initialTime: const Duration(seconds: 21),
                minTime: const Duration(seconds: 15),
                maxTime: const Duration(seconds: 30),
                baseUnit: BaseUnit.second,
              );
              ScaffoldMessenger.of(context)
                  .showSnackBar(SnackBar(content: Text('Chose duration: $resultingDuration')));
            },
            tooltip: 'Popup Duration Picker',
            child: const Icon(Icons.add),
          )),
    );
  }
}


```



CHANGELOG.md
## 2.0.2
* Add minTime,maxTime
* adjust padding to 32
* use default theme dialog
* enable HapticFeedback

## 2.0.0
* lints 2.0.1

## 1.1.0+1
#### New Features
* It is now possible to select the base unit of the timer using baseunit: parameter. 
See example app for an example.
  
## 1.0.1
#### Bug fixes
* Updated example app

## 1.0.0
#### Breaking Changes
* Added null-safety
#### New Features
* Popup duration picker with box decoration
#### Bug fixes
* Fixed Method not found: 'ButtonTheme.bar'

# Changelog from flutter_duration_picker
## [1.0.4] - Oct 23, 2018

* Fixed bug showing 60mins when approaching 0.

### [1.0.3] - Aug 01, 2018

* Fixed bug returning false from void function

## [1.0.2] - Jul 25, 2018

* Fixed initial duration when exceeding 1h

## [1.0.1] - Jun 26, 2018	 

* initial release.



example/lib/main.dart
import 'package:duration_picker2/duration_picker.dart';
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Duration Picker Demo',
      theme: ThemeData(
        useMaterial3: true,
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Duration Picker Demo'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  Duration _initTime = const Duration(seconds: 21);
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Expanded(
                child: DurationPicker(
              initialTime: _initTime,
              minTime: const Duration(seconds: 15),
              maxTime: const Duration(seconds: 30),
              enableHapticFeedback: false,
              baseUnit: BaseUnit.second,
              onChange: (val) {
                setState(() {
                  _initTime = val;
                });
              },
              snapToMins: 5.0,
            ))
          ],
        ),
      ),
      floatingActionButton: Builder(
          builder: (BuildContext context) => FloatingActionButton(
                onPressed: () async {
                  var resultingDuration = await showDurationPicker(
                    context: context,
                    initialTime: const Duration(seconds: 21),
                    minTime: const Duration(seconds: 15),
                    maxTime: const Duration(seconds: 30),
                    baseUnit: BaseUnit.second,
                  );
                  ScaffoldMessenger.of(context)
                      .showSnackBar(SnackBar(content: Text('Chose duration: $resultingDuration')));
                },
                tooltip: 'Popup Duration Picker',
                child: const Icon(Icons.add),
              )),
    );
  }
}
